core-blocks/heading/index.js

```diff
@@ -1,16 +1,60 @@
+/**
+ * External dependencies
+ */
+import { omit } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
 import { concatChildren } from '@wordpress/element';
-import { createBlock, getPhrasingContentSchema } from '@wordpress/blocks';
+import {
+	createBlock,
+	getPhrasingContentSchema,
+	getBlockAttributes,
+	getBlockType,
+} from '@wordpress/blocks';
 import { RichText } from '@wordpress/editor';
 
 /**
  * Internal dependencies
  */
 import edit from './edit';
 
+/**
+ * Given a node name string for a heading node, returns its numeric level.
+ *
+ * @param {string} nodeName Heading node name.
+ *
+ * @return {number} Heading level.
+ */
+export function getLevelFromHeadingNodeName( nodeName ) {
+	return Number( nodeName.substr( 1 ) );
+}
+
+const supports = {
+	className: false,
+	anchor: true,
+};
+
+const schema = {
+	content: {
+		type: 'array',
+		source: 'children',
+		selector: 'h1,h2,h3,h4,h5,h6',
+	},
+	level: {
```

Comment:
Changing block attributes (nodeName replaced by level) is a breaking change and until now we don't warn about it (`deprecated` call). This broke one of my plugins, I wonder if we can/should ensure Backwards compatibility here.

Reply:
>Changing block attributes (nodeName replaced by level) is a breaking change and until now we don't warn about it (`deprecated` call).

I'm not sure I totally understand. The block attributes are a public API we should be guaranteeing compatibility for? Is this to do with block transforms? There is a `deprecated` definition, though I sense it would not be enough for transforms (at least as currently implemented, though maybe possible to implement).

Reply:
What I'm saying is should we support both `nodeName` and `level` for some time coupled with `deprecated`? which hasn't been done in this PR (and it's not the only PR, just happened to be the one I noticed a bug with)

Reply:
I understand the breakage and potential remedy, just trying to think through how this is communicated: When deprecating a block attribute, is it not enough for a developer to be confident in using `deprecated`, as this is [all that we currently document in the handbook](https://wordpress.org/gutenberg/handbook/block-api/deprecated-blocks/#changing-the-attributes-set) ? Does this apply to all blocks, or just core blocks ?